_posts/2023-01-12-January-Update-Finishing-my-Igalia-CE.md

```diff
@@ -0,0 +1,228 @@
+userspace use of the ioctls in order to create useful tests. While I was
+exploring the v3dv, I was able to make two very simple contributions to Mesa:
+fixing typos and initializing a variable in order to assure proper error
+handling.
+
+| Patch | Status |
+| :---- | :----: |
+| [v3dv: fix multiple typos](https://gitlab.freedesktop.org/mesa/mesa/-/commit/d34f3a1db594c778e0c6bae7a5798742edb9635d)                               | **Accepted** |
+| [v3dv: initialize fd variable for proper error handling](https://gitlab.freedesktop.org/mesa/mesa/-/commit/a2252adde8235d5c4d78d9347527cd7914bb905a) | **Accepted** |
+
+# IGT tests to VC4
+---
+
+Although my main focus was on the V3D tests, as the VC4 and V3D drivers are kind
+of similar, I was able to bring some improvements to the VC4 tests as well. I
+added tests for perfmons and the `vc4_mmap` ioctl and improved a couple of
+things in the tests, such as moving a separate folder and creating a check to
+skip the VC4 tests if they are running on a Raspberry Pi 4.
+
+| Patch/Series | Status |
+| :---- | :----: |
+| [VC4 IGT Tests Updates](https://patchwork.freedesktop.org/series/110948/)                             | **Accepted** |
+| [Tests for V3D/VC4 Mmap BO IOCTLs](https://patchwork.freedesktop.org/series/112215/)                  | **Accepted** |
+| [Make sure v3d/vc4 support performance monitor](https://patchwork.freedesktop.org/series/112370/)     | In Review |
+| [tests/vc4_purgeable_bo: Fix conditional assertion](https://patchwork.freedesktop.org/patch/516737/)  | In Review |
+
+## Linux Kernel
+
+Although I didn't explore the VC4 driver as much as the V3D driver, I also took
+a look at the driver, and I was able to detect a small thing that could be
+improved: using the DRM-core helpers instead of open-code. Moreover, after a
+report on the mailing list, I bisected a deadlock and I was able to fix it after
+some study about the KMS locking system.
+
+| Patch | Status |
+| :---- | :----: |
+| [drm/vc4: drop all currently held locks if deadlock happens](https://patchwork.freedesktop.org/series/112299/)                                    | **Accepted** |
+| [drm/vc4: replace drm_gem_dma_object for drm_gem_object in vc4_exec_info](https://patchwork.freedesktop.org/patch/516491/?series=112347&rev=1)    | In Review |
+| [drm/vc4: replace obj lookup steps with drm_gem_objects_lookup](https://patchwork.freedesktop.org/patch/516490/?series=112347&rev=1)              | In Review |
+
+# The debugfs side-quest
+---
+The debugfs side-quest was a total coincidence during this project. I had some
+spare time and was looking for something to develop. While looking at the DRM
+TODO list, I bumped into the debugfs clean-up task and found it interesting to
+work on. So, I started to work on this task based on the previous work from
+Wambui Karuga and I talked to Melissa about it. By chance, Melissa was involved
+in Wambui's internship and she was able to help me figure out the last pieces of
```

Comment:
No, I wasn't involved in Wambui's internship, but in the Beatriz internship who also worked on this code